audio-worklet/src/noisegate-script-processor.js

```diff
@@ -95,18 +107,12 @@ class NoiseGate {
     }
 
     let envelope = this.detectLevel_(this.channel_);
-
-    // For sine waves, the envelope eventually reaches an average power of
-    // a^2 / 2. Sine waves are therefore scaled back to the original amplitude,
-    // but other waveforms or constant sources can only be approximated.
-    for (let k = 0; k < this.envelope_.length; k++) {
-      envelope[k] =  this.toDecibel_(Math.sqrt(envelope[k]) * Math.sqrt(2));
-    }
+    envelope = this.convertEnvelopeToSineNormalizedDecibelScale_(envelope);
 
     for (let i = 0; i < inputBuffer.numberOfChannels; i++) {
       let input = inputBuffer.getChannelData(i);
       let output = event.outputBuffer.getChannelData(i);
-      let weights = this.computeGain_(envelope);
+      let weights = this.computeWeights_(envelope);
```

Comment:
I moved the weight computation outside the loop, but I left the scaling in a separate function. 
I think its cleaner if computeWeights_ function does not alter its input when setting the weights. I also independently test computeWeights_ in the test code, and modifying the envelope in the computeWeights_ function would obfuscate that code.